hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/helpers/ChunkUtils.java

```diff
@@ -256,12 +257,21 @@ public static boolean validateChunkForOverwrite(File chunkFile,
   public static boolean isOverWriteRequested(File chunkFile, ChunkInfo
       chunkInfo) {
 
-    if (!chunkFile.exists()) {
+    long offset = chunkInfo.getOffset();
+    try {
+      FileInputStream inputStream = new FileInputStream(chunkFile);
+      inputStream.getChannel().position(offset);
```

Comment:
Can we use `chunkFile.length()` instead of seek and read?

Reply:
chunkfile.length depend on the offset

Reply:
If my understanding is correct, you are trying to solve sparse file "overwrite" problem.

But this code still prints true.

```java
    public static void main(String[] args) {
        File f = new File("/tmp/hello");
        try (FileOutputStream out = new FileOutputStream(f);
             FileInputStream in = new FileInputStream(f)) {

            out.getChannel().position(8 << 10); // 8KB
            out.write("hello, world".getBytes(StandardCharsets.UTF_8));

            System.out.println(f.length()); // prints 8204 (= 8192 + 12)

            in.getChannel().position(4 << 10); // 4KB
            if (in.read() == -1) { // read() = 0
                System.out.println(false);
            } else {
                System.out.println(true); // prints true
            }
        } catch (IOException e) {
            System.out.println(e);
        }
    }
```

Reply:
Ozone write the contents of the buffer from the current position to the limit

Reply:
Can you give an example of inputs, when your code and the original code will return different results?